src/main/java/io/neonbee/endpoint/metrics/NeonBeePrometheusHandler.java

```diff
@@ -0,0 +1,56 @@
+package io.neonbee.endpoint.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.neonbee.logging.LoggingFacade;
+import io.prometheus.client.exporter.common.TextFormat;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.micrometer.backends.BackendRegistries;
+import io.vertx.micrometer.impl.PrometheusScrapingHandlerImpl;
+
+/**
+ * A Vert.x Web {@link io.vertx.ext.web.Route} handler for Prometheus metrics scraping.
+ *
+ * The original Implementation doesn't work with {@link CompositeMeterRegistry}. This implementation fixes this.
+ */
+public class NeonBeePrometheusHandler extends PrometheusScrapingHandlerImpl {
+    private static final LoggingFacade LOGGER = LoggingFacade.create();
+
+    private final String registryName;
+
+    /**
+     * Constructs a new instance of NeonBeePrometheusHandler.
+     *
+     * @param registryName The name of the micrometer registry
+     */
+    public NeonBeePrometheusHandler(String registryName) {
+        super(registryName);
+        this.registryName = registryName;
+    }
+
+    @Override
+    public void handle(RoutingContext rc) {
+        MeterRegistry meterRegistry;
+        if (registryName == null) {
+            meterRegistry = BackendRegistries.getDefaultNow();
+        } else {
+            meterRegistry = BackendRegistries.getNow(registryName);
+        }
+
+        if (meterRegistry instanceof CompositeMeterRegistry) {
+            CompositeMeterRegistry cmr = (CompositeMeterRegistry) meterRegistry;
+            cmr.getRegistries().stream().filter(registry -> registry instanceof PrometheusMeterRegistry).findAny()
+                    .map(PrometheusMeterRegistry.class::cast)
+                    .ifPresentOrElse(pmr -> handleWithPrometheusMeterRegistry(rc, pmr), () -> LOGGER
```

Comment:
in case of error we need to fail the rc, otherwise request hangs forever.

Reply:
I have fixed this issue.